src/main/java/teammates/ui/webapi/GetNotificationAction.java

```diff
@@ -0,0 +1,53 @@
+package teammates.ui.webapi;
+
+import java.util.List;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.FieldValidator;
+import teammates.ui.output.NotificationData;
+import teammates.ui.output.NotificationsData;
+import teammates.ui.request.InvalidHttpRequestBodyException;
+
+/**
+ * Action: Get a list of notifications.
+ */
+public class GetNotificationAction extends Action {
+
+    @Override
+    AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    void checkSpecificAccessControl() throws UnauthorizedAccessException {
+        // Any user can get notifications as long as its parameters are valid
+    }
+
+    @Override
+    public JsonResult execute() throws InvalidHttpRequestBodyException, InvalidOperationException {
+        String targetUser = getRequestParamValue(Const.ParamsNames.NOTIFICATION_TARGET_USER);
+        //        boolean isFetchingAll = Boolean.parseBoolean(
```

Comment:
Can remove if not needed anymore

Reply:
Seems like this is code for after we start implementing read status? Would be good to add a TODO here